### PR TITLE
Varvara/feature/godaddy integration

### PR DESCRIPTION
**Pull request by varvarashcherbyna7**

### CODE REVIEW

File: naples/routes/auth.py [hunk 1]
```diff
@@ -156,6 +157,22 @@ def sign_up(
...
+    dns_record = s.DNSRecord(
+        domain=CFG.MAIN_DOMAIN,
+        subdomain=new_user.uuid,
+        record_type=CFG.RECORD_TYPE,
+        ttl=CFG.GO_DADDY_TTL,
+        value=CFG.GODADDY_IP_ADDRESS,
+        api_url=CFG.GODADDY_API_URL,
+        api_key=CFG.GODADDY_API_KEY,
+        api_secret=CFG.GODADDY_API_SECRET,
+    )
```
Comment (Taraslysak) [inline naples/routes/auth.py]: This can be hidden inside `add_dns_record` function

File: services/store/add_dns_record.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from naples.config import config
+
+CFG = config()
+
+
+def check_subdomain_existence(subdomain, headers, url):
+    response = requests.get(url, headers=headers)
+    response.raise_for_status()
+
+    dns_records = response.json()
```
Comment (Taraslysak) [inline services/store/add_dns_record.py:14]: Add pydantic validation here

File: services/store/add_dns_record.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    url = f"{record.api_url}/domains/{record.domain}/records"
+
+    log(log.INFO, "URL: [%s]", url)
+
+    headers = {
+        "Authorization": f"sso-key {record.api_key}:{record.api_secret}",
+        "Content-Type": "application/json",
+    }
+
+    log(log.INFO, "Headers: [%s]", headers)
```
Comment (Taraslysak) [inline services/store/add_dns_record.py]: Please remove this log. We do not want sensitive API data do be exposed in logs

File: services/store/add_dns_record.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    return False  # Subdomain not found or not of type A
+
+
+def add_dns_record(record: s.DNSRecord):
+    # URL for the GoDaddy API to add a DNS record
+
+    url = f"{record.api_url}/domains/{record.domain}/records"
+
+    log(log.INFO, "URL: [%s]", url)
```
Comment (Taraslysak) [inline services/store/add_dns_record.py]: Please, make you logs more descriptive. at least add function name into logs, so we could better understand it when we will read the logs

File: services/store/add_dns_record.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    payload = [
+        {
+            "type": record.record_type,
+            "name": record.subdomain,
+            "data": record.value,
+            "ttl": record.ttl,
+        }
+    ]
+
+    log(log.INFO, "Payload: [%s]", payload)
```
Comment (Taraslysak) [inline services/store/add_dns_record.py]: This can be DEBUG

**Review by Taraslysak — COMMENTED** [review]
Please do some refactoring. 

Please also do not forget to handle store urls update in the following PRs, since this is VERY IMPORTANT

File: services/store/add_dns_record.py (hunk 5 above)
Comment (varvarashcherbyna7) [reply services/store/add_dns_record.py]: thanks for the comments, I'll fix it )

**Pull request merged**